src/nanotron/models/base.py

```diff
@@ -99,6 +99,39 @@ def log_modules(self, level: int = logging.DEBUG, group: Optional[ProcessGroup]
                 rank=rank,
             )
 
+    @property
+    def named_modules_in_pp_rank(self) -> Dict[str, nn.Module]:
+        """Return the named modules that only belongs to the current pp rank.
+
+        An example output:
+        {
+            'module_name': module,
+            ...
+        }
+
+        NOTE: not include module_name.weight or bias, but only module_name
+        """
+
+        def get_leaf_modules(module: nn.Module) -> List[Tuple[str, nn.Module]]:
+            """
+            Return all the leaf modules (modules without any child modules) in a PyTorch module.
+            """
+            leaf_modules = []
+            for n, m in module.named_modules():
+                if not list(m.children()):
+                    leaf_modules.append((n, m))
+            return leaf_modules
```

Comment:
i think we can use this 

https://github.com/huggingface/nanotron/blob/b7e6a1c079e8298456897be03b8d15b0854e94f7/src/nanotron/parallel/sharded_parameters.py#L113-L116